synapse/tools/axon/load.py

```diff
@@ -0,0 +1,112 @@
+import os
+import sys
+import asyncio
+import hashlib
+
+import synapse.exc as s_exc
+import synapse.common as s_common
+import synapse.telepath as s_telepath
+
+import synapse.lib.cmd as s_cmd
+import synapse.lib.coro as s_coro
+import synapse.lib.output as s_output
+import synapse.lib.msgpack as s_msgpack
+
+descr = '''
+Load blobs into a Synapse Axon.
+'''
+
+async def loadBlobs(opts, outp, blobsfiles):
+    try:
+        async with await s_telepath.openurl(opts.url) as axon:
+
+            cellinfo = await axon.getCellInfo()
+            celltype = cellinfo['cell']['type']
+            if "axon" not in celltype.lower():
+                mesg = f'Axon load tool only works on axons, not {celltype}'
+                raise s_exc.TypeMismatch(mesg=mesg)
+
+            for blobsfile in blobsfiles:
+                outp.printf(f'Processing blobs file: {blobsfile}')
+                with open(blobsfile, 'rb') as fd:
+                    msgit = s_msgpack.iterfd(fd)
+                    for mesg in msgit:
+                        match mesg:
+                            case ("blob:init", meta):
+                                outp.printf(f"Loading blobs file: {meta}")
+                                continue
+                            case ("blob:fini", meta):
+                                outp.printf(f"Finished loading: {meta}")
+                                break
+                            case ("blob", meta):
+                                sha2hex = meta['sha256']
+                                size = meta['size']
+                                sha256 = s_common.uhex(sha2hex)
+                                async with await axon.upload() as upfd:
+                                    total = 0
+                                    hasher = hashlib.sha256()
+                                    while True:
+                                        try:
+                                            byts = next(msgit)
+                                        except StopIteration:
+                                            mesg = f'Unexpected end of file while reading blob {sha2hex}'
+                                            raise s_exc.BadDataValu(mesg=mesg)
+                                        if type(byts) is tuple:
+                                            msgit = (i for i in [byts] + list(msgit))
+                                            break
```

Comment:
I don't understand this block of code? Are you effectively trying to put the message back in the iterator because it's a different message type? I think we should talk about the "protocol" of these blob files.